### PR TITLE
Fix verify-vendor script to check all go.mod and go.sum files

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Fixes the verify-vendor script to check all go.mod/go.sum/go.work files are correct

Includes the update-vendor commit from https://github.com/kubernetes/kubernetes/pull/126349 to correct go.mod / go.sum files and unbreak the publishing bot

xref https://github.com/kubernetes/kubernetes/pull/126349

```release-note
NONE
```

/area code-organization
/assign @dims

### CODE REVIEW

**Comment by liggitt** [comment]
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/126353/pull-kubernetes-dependencies/1816466657804029952 failed as expected with the current diff, so master is protected again

**Comment by dims** [comment]
cc @kubernetes/release-team

**Comment by liggitt** [comment]
This has to be included in 1.31 so the publishing bot and go.mod / go.sum files are correct

File: staging/src/k8s.io/dynamic-resource-allocation/go.mod [hunk 1]
```diff
@@ -82,6 +84,7 @@ replace (
 	k8s.io/apiserver => ../apiserver
 	k8s.io/client-go => ../client-go
 	k8s.io/component-base => ../component-base
+	k8s.io/component-helpers => ../component-helpers
```
Comment (akhilerm) [inline staging/src/k8s.io/dynamic-resource-allocation/go.mod:87]: q: Does the update-vendor script takes care of adding this replace automatically, or its a manual change?

Comment (liggitt) [reply staging/src/k8s.io/dynamic-resource-allocation/go.mod:87]: update script adds this

**Comment by liggitt** [comment]
shellcheck wanted a variable in the verify script quoted, repushed

**Review by akhilerm — APPROVED** [review]
/lgtm

**Comment by MadhavJivrajani** [comment]
@liggitt @dims let's also backport atleast the verify script changes?

**Comment by akhilerm** [comment]
> @liggitt @dims let's also backport atleast the verify script changes?

Isnt it needed only in release branches that were created after the go workspace change (release-1.30) ?

**Pull request merged**